extensions/2.0/Khronos/KHR_materials_diffuse_transmission/README.md

```diff
@@ -0,0 +1,241 @@
+
+<figure style="text-align:center">
+<p float="left">
+<img src="figures/leaves.jpg" height="350" />
+<img src="figures/thin_translucent.jpg" height="350" />
+</p>
+<sub><figcaption><em>(Thin) translucent leaves/foliage</em></figcaption></sub>
+</figure>
+
+## Extending Materials
+
+The effect is activated by adding the `KHR_materials_diffuse_transmission` extension to any glTF material.
+
+```json
+{
+    "materials": [
+        {
+            "extensions": {
+                "KHR_materials_diffuse_transmission": {
+                    "diffuseTransmissionFactor": 0.25,
+                    "diffuseTransmissionTexture": {
+                      "index": 0
+                    },
+                    "diffuseTransmissionColorFactor": [
+                      1.0,
+                      0.9,
+                      0.85
+                    ]
+                }
+            }
+        }
+    ]
+}
+```
+
+|                                     | Type                                                                | Description                                                                                                                                                                                         | Required                 |
+| ----------------------------------- | ------------------------------------------------------------------- | --------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ------------------------ |
+| **diffuseTransmissionFactor**       | `number`                                                            | The percentage of reflected, non-specularly reflected light that is transmitted through the surface via the Lambertian diffuse transmission, i.e., the strength of the diffuse transmission effect. | No, default: `0`         |
+| **diffuseTransmissionTexture**      | [`textureInfo`](/specification/2.0/README.md#reference-textureInfo) | A texture that defines the strength of the diffuse transmission effect, stored in the alpha (A) channel. Will be multiplied by the diffuseTransmissionFactor.                                       | No                       |
+| **diffuseTransmissionColorFactor**  | `number[3]`                                                         | The color of the transmitted light.                                                                                                                                                                 | No, default: `[1, 1, 1]` |
```

Comment:
I would say that this is the color of the re-emitted light.
`The color of the re-emitted light.`

Reply:
Well, it's the same when its called `color of the transmitted light` - it still depends on the received light.....
I think that something along the lines of:
`Internal coloring of the light that is transmitted through the surface`

Reply:
How about `The color that modulates the transmitted light`?

Reply:
I really think we must re-work this extension and word it in a more physically based manner.
The phenomena that is modelled is attenuation of light - we have simplified it so that light is directly attenuated when entering the material.
On the opposite side there will be reflection and transmission as light hits the surface.
Of the transmitted light (from the opposite side) we have a factor controlling the amount that will enter into the surface - all of this light will be attenuated (regardless of distance) and re-emitted.

As a final parameter we can add the so-called thickness texture - in this case it is more of an absorption texture.
This texture controlls the amount of attenuated light that is re-emitted, regardless of thickness, we really do not know if light is absorbed due to distance travelled or density of material.
And so it ought to be called `absorptionTexture` and can work the same way that occlussion texture works (ie higher values let light trough, lower values reduce)

Reply:
Not sure if I really understand your thickness texture proposal. Isn't it exactly what you control with the intensity of the diffuseTransmissionColorTexture?

Reply:
Apologies if my own demos have led to some of the confusion here.  Both transmission and diffuse transmission are surface-only properties.  When used on a thin-walled surface, either of these properties allows light to transmit directly to the opposite side.  But, if the volume extension is present, there may be a thick volume beyond the surface.  In that case, transmission or diffuse transmission is just a gateway allowing light to enter and exit the volume.  The properties of T or DT don't affect the volume itself, as the volumetric properties are covered by its own extension (as thickness, attenuation color, attenuation distance).

I gave a demo of [candle wax](https://github.com/KhronosGroup/glTF-Sample-Viewer/pull/544#issuecomment-2177354713) recently.  In the material for this, diffuse transmission is set to 0.5, untextured white.  All of the wax coloring is established by volumetric properties (thickness and attenuation), and DT is just an enabler for the light to get in and out of that volume.  You can imagine that DT offers a surface-level scattering event, white in this case, paired with a volumetric attenuation towards wax color.

```json
    "materials":[
        {
            "doubleSided":false,
            "extensions":{
                "KHR_materials_diffuse_transmission":{
                    "diffuseTransmissionFactor":0.5
                },
                "KHR_materials_volume":{
                    "attenuationColor":[
                        0.8964201807975769,
                        0.8795840740203857,
                        0.6938217878341675
                    ],
                    "attenuationDistance":0.02,
                    "thicknessFactor":0.11999999731779099,
                    "thicknessTexture":{
                        "index":0
                    }
                },
                "KHR_materials_ior":{
                    "ior":1.4500000476837158
                }
            },
            "name":"Wax Material",
            "occlusionTexture":{
                "index":1
            },
            "pbrMetallicRoughness":{
                "baseColorFactor":[
                    0.8964201807975769,
                    0.8795840740203857,
                    0.6938217878341675,
                    1
                ],
                "metallicFactor":0,
                "roughnessFactor":0.30000001192092896
            }
        },
```

Something similar happens in my dragon diffuse transmission demo, as the dragon's coloring in that demo comes from the volumetric properties as well.  I don't yet have a demo that pairs surface-level coloring with volumetric-imposed coloring, but the two are mathematically separate: The diffuse transmission coloring would be along the surface, and would not change with thickness.  Volumetric attenuation is an un-texturable homogeneous color throughout the volume, that does potentially change with thickness.  Light reaching the eye must survive both.